shell/platform/darwin/ios/framework/Source/spring_curve_ios.mm

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/spring_curve_ios.h"
+
+#include <Foundation/Foundation.h>
+
+// Spring calculation adapted from Pan Yusheng's research project.
+// See: https://github.com/CosynPa/RevealSpringAnimation.
```

Comment:
Is there a specific file this is derived from?

In general there's a concern here about copyright since Pan's repo doesn't seem to have a license and Pan themselves haven't commented on this PR granting us the rights to their code, so if it's a direct transliteration it's not something we'd be comfortable with but if it's merely broadly inspired by their work then that would be fine. If it's more like the former then we would want to ask Pan for explicit permission and for them to sign the CLA.

Reply:
Sorry, didn't think about that.

The spring calculation found in this file is a direct transliteration from Pan's swift code to Objc. Pan credits [Framer](https://github.com/koenbok/Framer/blob/master/framer/Animators/SpringCurveValueConverter.coffee) for the spring calculation math which is under an MIT license. I don't know Pan nor have I communicated with him.

I'm unfamiliar with licensing considerations for open source software, but the underlying spring math seems to be commonly understood. Another reference (which is MIT licensed) is found in the [react codebase](https://github.com/facebook/react-native/blob/8bd3edec88148d0ab1f225d2119435681fbbba33/ReactAndroid/src/main/java/com/facebook/react/animated/SpringAnimation.java).

Let me know how you think we should move forward.

Reply:
Maybe emmmmm... we can't get a resp from him. I saw him is offline on github very long time😭😭.......

Reply:
Will take a swing at adapting the react spring calculation code tomorrow - hopefully it's portable and can produce similar results

Reply:
It doesnt seem that spring animation is restricted by copyright. Dont you noted that Pan himself is leveraging on the spring calculation from Framer, which is under MIT? As for the React spring animation, it should be the one for android platform, if so it may be tested for the android fix along with the other one from keyboard_attachable https://github.com/flutter/flutter/issues/116836#issuecomment-1353819652

Reply:
Since changed the spring model to avoid licensing issues